io-engine/src/rebuild/rebuild_log.rs

```diff
@@ -109,43 +84,58 @@ impl RebuildLog {
         (lbn * self.block_len / SEGMENT_SIZE) as usize
     }
 
-    /// Counts the total number of modified blocks.
-    pub fn count_modified_blocks(&self) -> u64 {
-        let segs = self.segments.iter().filter(|i| *i).count() as u64;
-        segs * SEGMENT_SIZE / self.block_len
+    /// Counts the total number of blocks set to one.
+    pub fn count_ones(&self) -> u64 {
+        self.segments.iter().filter(|i| *i).count() as u64
+    }
+
+    /// Counts the total number of dirty blocks.
+    pub fn count_dirty_blks(&self) -> u64 {
+        self.count_ones() * SEGMENT_SIZE / self.block_len
     }
 }
 
-/// A handle object for a rebuild log instance.
-/// A rebuild log instance allows to have multiple handle to access it.
-#[derive(Clone)]
-pub struct RebuildLogHandle {
-    /// Share log reference.
-    log: Arc<Mutex<RebuildLog>>,
-    /// Name of the underlying device.
-    /// Keep it here to avoid locking just to access the device name,
-    /// and cloning the name when accessing it.
+/// Per-core I/O log channel.
+/// I/O log channel is enables lockless logging of I/O operations.
+pub struct IOLogChannelInner {
```

Comment:
Should this part of the rebuild be inside rebuild_log module? The rebuild itself only seems to care about the combined segment rebuildMap?

Reply:
Yes, true. That was actually my intent: I renamed non-rebuild part to "I/O log". Non-rebuild part has to be moved into nexus part, indeed. Will do.